src/datepicker/datepicker.spec.ts

```diff
@@ -23,13 +26,43 @@ function getDatepicker(element: HTMLElement): HTMLElement {
   return element.querySelector('ngb-datepicker') as HTMLElement;
 }
 
+function expectSameValues(datepicker: NgbDatepicker, config: NgbDatepickerConfig) {
+  expect(datepicker.dayTemplate).toBe(config.dayTemplate);
+  expect(datepicker.firstDayOfWeek).toBe(config.firstDayOfWeek);
+  expect(datepicker.markDisabled).toBe(config.markDisabled);
+  expect(datepicker.minDate).toBe(config.minDate);
+  expect(datepicker.maxDate).toBe(config.maxDate);
+  expect(datepicker.showNavigation).toBe(config.showNavigation);
+  expect(datepicker.showWeekdays).toBe(config.showWeekdays);
+  expect(datepicker.showWeekNumbers).toBe(config.showWeekNumbers);
+  expect(datepicker.startDate).toBe(config.startDate);
+}
+
+function customizeConfig(config: NgbDatepickerConfig) {
+  config.dayTemplate = {} as TemplateRef<DayTemplateContext>;
+  config.firstDayOfWeek = 2;
+  config.markDisabled = (date) => false;
+  config.minDate = {year: 2000, month: 0, day: 1};
+  config.maxDate = {year: 2030, month: 1, day: 31};
```

Comment:
If I read the code correctly:

By default, if left undefined, the min date is computed by using the 1st of January of ten years before the start date (or now if there is no start date).

By default, if left undefined, the max date is computed by using the 31 of December of ten years after the start date (or now if there is no start date).

@maxokorokov can you confirm?

But this is done by the Calendar abstraction, which by default is Gregorian, but can be changed to another implementation. So I think we're fine here. 

Note that the above function is just used in unit tests to check that custom config values are honored by the component. I didn't change anything to the min/max date selection logic.